### PR TITLE
FEDX-1670: Fixed getters, setters, and field relationships

### LINKED ISSUE #119: Setters generate invalid scip symbol
Given the following class, this is the snapshot that would be generated
```dart
class Foo {
  int _value = 0;

  get value => _value
//    ^^^^^ definition ... Foo#value

  set value(int newValue) => _value = newValue
//    ^^^^^ definition ... Foo#value=.
}
```

Sourcegraph treats symbols with the suffix `=.` as invalid and will error:

Screenshot 2024-02-07 at 2 59 03 PM

`scip-typescript` treats setters/getters the following way:
```ts
class Foo {
    private _value: number;

    public get value() {
//             ^^^^^ definition ... Foo#<get>value.
        return this._value;
    }

    public set value(value: number) {
//             ^^^^^ definition ... Foo#<set>value.
        this._value = value;
    }
}


// Note, that `+=` has both a reference to get and set
new Foo().value += 1;
//        ^^^^^ reference ... Foo#`<get>value`().
//        ^^^^^ reference ... Foo#`<set>value`().
```

we should mirror this pattern for scip-dart

### LINKED ISSUE #151: Relationships not calculated on fields
```dart
abstract class Foo {
  String name = '';
}

class Bar extends Foo {
  @override
  String name = 'asdf';
//       ^^^^ Does not contain a relationship to Foo.name
}
```

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
# [FEDX-1670](https://jira.atl.workiva.net/browse/FEDX-1670)
Issue Status

getters, setters, and fields were not correctly indexing their relationships. This PR fixes that (Closes #151)

Also closes #119, because that was necessary in order to get the relationships fixed

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: snapshots/output/basic-project/lib/relationships.dart [hunk 1]
```diff
@@ -1,9 +1,16 @@
...
 //  ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
-//             ^^^^^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#hierarchy.
+//             ^^^^^^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#`<get>someGetter`.
+    set someSetter(String v);
+//      ^^^^^^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#`<set>someSetter`.
+//                 ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
+//                        ^ definition local 0
+    String field = '';
+//  ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
+//         ^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#field.
```
Comment (matthewnitschke-wk) [inline snapshots/output/basic-project/lib/relationships.dart:13]: `Mammal` has sub-classes extending it, but it is, itself, at the top level

All fields, methods, getters and setters, should not contain references, like is shown here

File: snapshots/output/basic-project/lib/relationships.dart [hunk 2]
```diff
@@ -13,6 +20,13 @@
     String sound() => 'NOISE!';
 //  ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
 //         ^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Animal#sound().
+  
+    @override
+//   ^^^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`annotations.dart`/override.
+    String field = 'asdf';
+//  ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
+//         ^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Animal#field.
+//         relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#field. implementation reference
```
Comment (matthewnitschke-wk) [inline snapshots/output/basic-project/lib/relationships.dart:29]: `field` is overriding the same field in Mammal, relationship is correctly recorded

File: snapshots/output/basic-project/lib/relationships.dart [hunk 3]
```diff
@@ -29,6 +43,14 @@
 //      relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/SwimAction# implementation
 //                  ^^^^^^ reference scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Animal#
 //                              ^^^^^^^^^^ reference scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/SwimAction#
+    @override
+//   ^^^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`annotations.dart`/override.
+    String field = 'otherVal';
+//  ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
+//         ^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Dog#field.
+//         relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Animal#field. implementation reference
+//         relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#field. implementation reference
```
Comment (matthewnitschke-wk) [inline snapshots/output/basic-project/lib/relationships.dart:52]: `field` is overriding the same field both `Animal` AND `Mammal`, relationship is correctly recorded

File: snapshots/output/basic-project/lib/relationships.dart [hunk 4]
```diff
@@ -38,8 +60,16 @@
   
     @override
 //   ^^^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`annotations.dart`/override.
-    String get hierarchy => 'dog.animal.mammal';
+    String get someGetter => 'value';
 //  ^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`string.dart`/String#
-//             ^^^^^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Dog#hierarchy.
-//             relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#hierarchy. implementation reference
+//             ^^^^^^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Dog#`<get>someGetter`.
+//             relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#`<get>someGetter`. implementation reference
```
Comment (matthewnitschke-wk) [inline snapshots/output/basic-project/lib/relationships.dart:66]: getter correctly declared with `<get>` syntax, and relationship set correctly

File: snapshots/output/basic-project/lib/relationships.dart [hunk 5]
```diff
@@ -38,8 +60,16 @@
...
+  
+    @override
+//   ^^^^^^^^ reference scip-dart pub dart:core 2.19.0 dart:core/`annotations.dart`/override.
+    set someSetter(String v) {};
+//      ^^^^^^^^^^ definition scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Dog#`<set>someSetter`.
+//      relationship scip-dart pub dart_test 1.0.0 lib/`relationships.dart`/Mammal#`<set>someSetter`. implementation reference
```
Comment (matthewnitschke-wk) [inline snapshots/output/basic-project/lib/relationships.dart:72]: setter correctly declared with `<set>` syntax, and relationship set correctly

**Review by nicbiggs-wk — APPROVED** [review]

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x] Snapshots look good, and tests pass

🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**